### PR TITLE
Roadmap draft

### PR DESCRIPTION
**Pull request by thomaseizinger**
This PR adds a roadmap draft. Feel free to point out problems, issues or more ideas that could be added. However, the idea would be to for now just add this document and make changes to it once we get a clearer picture of what we want to do.

Changes request in this PR should therefore focus on "no-go" things or completely different view points of a particular topic.

### CODE REVIEW

**Comment by duffleit** [comment]
I really like this roadmap, but shouldn't it be created as an issue?

**Comment by thomaseizinger** [comment]
Actually, I created it as a document on purpose because it is not concrete enough in my opinion for an issue. I would much rather use issues for concrete tasks, feature requests or bugs instead of ideas. With ongoing edits and refinements, certain sections of the roadmap might be concrete enough to actually describe the feature or change detailed in an issue.
The thing I would not like about an issue or several issues in order to discuss this is that these issues are never really closed and I don't like unfinished things.

**Comment by duffleit** [comment]
I've seen it a couple of times that roadmaps and milestones are discussed though issues. Therefore I think it would be okay, but we should declare it accordingly, and provide links to more concrete issues in the following.

**Comment by thomaseizinger** [comment]
The problem I have with using issues for these sort of discussions is that they get bloated really quickly. I'd therefore much rather prefer sticking with a document that gets updated over time. Specific updates can then be discussed in issues and submitted via pull-requests.

**Comment by duffleit** [comment]
Ok, as I don't have any experience with this, I am neither for nor against any of theses options. It is best to try it out and see how it feels. Let's start with it as a document! 👍

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Roadmap
+
+This document summarizes ideas and tries to group them into categories / topics. This list is unordered.
+The purpose of this document is to gather ideas and refine requirements out of them in order to finally create issues as specific tasks for implementation.
+
+## Deleting / Editing bills
+
+Deleting or editing bills is a major requirement that has to implemented in one way or another. Due to the lack of accounts, it is hard to implement privileged functionality that would allow destructive actions. The design-goal for these features therefore has to be "always undo" and should allow every user to always undo any action that has been executed.
```
Comment (duffleit) [inline ROADMAP.md]: to be implemented

File: ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+- Deleting a bill: This should not really delete a bill but rather "hide" it from the nobt or mark it as old / unnecessary
+- Editing a bill: Every edit of a bill should result in a new version and every version should be restore-able. Another idea would be to lock certain elements from being changed (maybe only one thing at a time?)
+
+## Balances and "tickable" transactions
+
+Currently, the final transactions that describe all debts are transient, i.e. calculated on-the-fly when the nobt is retrieved from the server. This means, that the individual transactions are not addressable and therefore, no state can be saved for them. In order to simplify the whole settling process, a feature called "freeze nobt" could be introduced. Freezing a nobt would persists the current transactions into the database and prohibits further bills from being added to the nobt. Persisting the transactions to the database makes them addressable, which means that we can implement a UI that allows users to "tick-off" individual transactions and thereby mark them as paid.
```
Comment (duffleit) [inline ROADMAP.md]: would persist the

File: ROADMAP.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+A different way that would still allow to add further bills would be to mark all current bills as "processed" and merge the persisted transactions with the once resulting from the new bills. This would make it possible to tick-off individual transactions. However a major problem to be solved concerning UX is how to communicate to the user, why some transactions can be ticked off and some cannot. It is probably easier to understand, that no bills can be added until all the transactions are ticked off.
```
Comment (duffleit) [inline ROADMAP.md]: However, a

**Pull request merged**